dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.TextToImage.cs

```diff
@@ -47,4 +49,113 @@ internal async Task<string> GenerateImageAsync(
+    internal async Task<IReadOnlyList<ImageContent>> GetImageContentsAsync(
+        TextContent input,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        // Ensure the input is valid
+        Verify.NotNull(input);
+
+        // Convert the generic execution settings to OpenAI-specific settings
+        var imageSettings = OpenAITextToImageExecutionSettings.FromExecutionSettings(executionSettings);
+
+        var imageGenerationOptions = new ImageGenerationOptions()
+        {
+            Size = GetGeneratedImageSize(imageSettings.Size),
+            ResponseFormat = GetResponseFormat(imageSettings.ResponseFormat),
+            Style = GetGeneratedImageStyle(imageSettings.Style),
+            Quality = GetGeneratedImageQuality(imageSettings.Quality),
+            EndUserId = imageSettings.EndUserId,
+        };
+
+        var targetModel = string.IsNullOrEmpty(imageSettings.ModelId)
+            ? "dall-e-2" // Defaults to the DALL-E 2 server-side - https://platform.openai.com/docs/api-reference/images/create#images-create-model.
+            : imageSettings.ModelId;
+
+        ClientResult<GeneratedImage> response = await RunRequestAsync(() => this.Client!.GetImageClient(targetModel).GenerateImageAsync(input.Text, imageGenerationOptions, cancellationToken)).ConfigureAwait(false);
+        var generatedImage = response.Value;
+
+        List<ImageContent> result = [];
+        if (generatedImage.ImageUri is not null)
+        {
+            result.Add(new ImageContent(uri: generatedImage.ImageUri) { InnerContent = generatedImage });
+        }
+        else
+        {
+            result.Add(new ImageContent(generatedImage.ImageBytes, "image/png") { InnerContent = generatedImage });
+        }
+
+        return result;
+    }
+
+    private static GeneratedImageSize? GetGeneratedImageSize((int Width, int Height)? size)
+        => size is null
+            ? null
+            : new GeneratedImageSize(size.Value.Width, size.Value.Height);
+
+    private static GeneratedImageQuality? GetGeneratedImageQuality(string? quality)
+    {
+        if (quality is null)
+        {
+            return null;
+        }
+
+        return quality.ToUpperInvariant() switch
+        {
+            "STANDARD" => GeneratedImageQuality.Standard,
+            "HIGH" or "HD" => GeneratedImageQuality.High,
+            _ => throw new NotSupportedException($"The provided quality '{quality}' is not supported.")
+        };
+    }
+
+    private static GeneratedImageStyle? GetGeneratedImageStyle(string? style)
+    {
+        if (style is null)
+        {
+            return null;
+        }
+
+        return style.ToUpperInvariant() switch
+        {
+            "VIVID" => GeneratedImageStyle.Vivid,
+            "NATURAL" => GeneratedImageStyle.Natural,
+            _ => throw new NotSupportedException($"The provided style '{style}' is not supported.")
```

Comment:
Since we are using `OpenAITextToImageExecutionSettings`, should we just put these enums like `GeneratedImageStyle` and `GeneratedImageQuality` in `OpenAITextToImageExecutionSettings`, so we don't rely on strings? It should minimize user error. We can also create our own enums in abstractions if needed to work with images.

If not, then in each method that performs switch/case on these enums we should specify in exception message which values are supported for easier troubleshooting for users.

Reply:
Wil append some details from your comments in this open issue for further discussion.

- #8533